x-pack/plugins/fleet/server/services/epm/package_service.ts

```diff
@@ -203,6 +217,37 @@ class PackageClientImpl implements PackageClient {
     });
   }
 
+  public async installCustomIntegration(options: {
+    pkgName: string;
+    kibanaVersion?: string;
+    force?: boolean | undefined;
+    spaceId?: string | undefined;
+    datasets: CustomPackageDatasetConfiguration[];
+  }): Promise<InstallResult> {
+    await this.#runPreflight(INSTALL_PACKAGES_AUTHZ);
+
+    const {
+      pkgName,
+      kibanaVersion = appContextService.getKibanaVersion(),
+      datasets,
+      spaceId = DEFAULT_SPACE_ID,
+      force = false,
+    } = options;
+
+    return await installPackage({
+      force,
+      pkgName,
+      kibanaVersion,
+      datasets,
+      spaceId,
+      installSource: 'custom',
+      esClient: this.internalEsClient,
+      savedObjectsClient: this.internalSoClient,
+      neverIgnoreVerificationError: !force,
+      authorizationHeader: this.getAuthorizationHeader(),
+    });
+  }
```

Comment:
Fleet team: This new method allows installing custom integrations using the plugin's start contract. There is already a REST API to achieve this but the package service currently only allows installing packages from the registry.